metadata-ingestion/tests/unit/sql_parsing/test_schemaresolver.py

```diff
@@ -1,26 +1,64 @@
-from datahub.sql_parsing.schema_resolver import SchemaResolver, _TableName
+from datahub.sql_parsing.schema_resolver import (
+    SchemaInfo,
+    SchemaResolver,
+    _TableName,
+    match_columns_to_schema,
+)
 
 
-def test_basic_schema_resolver():
+def create_default_schema_resolver(urn: str) -> SchemaResolver:
     schema_resolver = SchemaResolver(
         platform="redshift",
         env="PROD",
         graph=None,
     )
 
     schema_resolver.add_raw_schema_info(
-        urn="urn:li:dataset:(urn:li:dataPlatform:redshift,my_db.public.test_table,PROD)",
+        urn=urn,
         schema_info={"name": "STRING"},
     )
 
+    return schema_resolver
+
+
+def test_basic_schema_resolver():
+    input_urn = (
+        "urn:li:dataset:(urn:li:dataPlatform:redshift,my_db.public.test_table,PROD)"
+    )
+
+    schema_resolver = create_default_schema_resolver(urn=input_urn)
+
     urn, schema = schema_resolver.resolve_table(
         _TableName(database="my_db", db_schema="public", table="test_table")
     )
-    assert (
-        urn
-        == "urn:li:dataset:(urn:li:dataPlatform:redshift,my_db.public.test_table,PROD)"
+
+    assert urn == input_urn
+
+    assert schema
+
+    assert schema["name"]
+
+    assert schema_resolver.schema_count() == 1
+
+
+def test_resolve_urn():
+    input_urn: str = (
+        "urn:li:dataset:(urn:li:dataPlatform:redshift,my_db.public.test_table,PROD)"
+    )
+
+    schema_resolver = create_default_schema_resolver(urn=input_urn)
+
+    schema_resolver.add_raw_schema_info(
+        urn=input_urn,
+        schema_info={"name": "STRING"},
```

Comment:
what does this test do that `test_basic_schema_resolver` does not do?

Reply:
This test is calling `resolve_urn` instead of `resolve_table`